libraries/ms-helper/src/mcp2515.c

```diff
@@ -330,3 +333,384 @@ StatusCode mcp2515_tx(Mcp2515Storage *storage, uint32_t id, bool extended, uint6
 
   return STATUS_CODE_OK;
 }
+#else
+#include <stddef.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "critical_section.h"
+#include "debug_led.h"
+#include "delay.h"
+#include "gpio_it.h"
+#include "log.h"
+#include "mcp2515.pb-c.h"
+#include "mcp2515_defs.h"
+#include "soft_timer.h"
+#include "store.h"
+#include "stores.pb-c.h"
+
+typedef struct Mcp2515TxBuffer {
+  uint8_t id;
+  uint8_t data;
+  uint8_t rts;
+} Mcp2515TxBuffer;
+
+typedef struct Mcp2515RxBuffer {
+  uint8_t id;
+  uint8_t data;
+  uint8_t int_flag;
+} Mcp2515RxBuffer;
+
+static const Mcp2515TxBuffer s_tx_buffers[] = {
+  { .id = MCP2515_LOAD_TXB0SIDH, .data = MCP2515_LOAD_TXB0D0, .rts = MCP2515_RTS_TXB0 },
+  { .id = MCP2515_LOAD_TXB1SIDH, .data = MCP2515_LOAD_TXB1D0, .rts = MCP2515_RTS_TXB1 },
+  { .id = MCP2515_LOAD_TXB2SIDH, .data = MCP2515_LOAD_TXB2D0, .rts = MCP2515_RTS_TXB2 },
+};
+
+static const Mcp2515RxBuffer s_rx_buffers[] = {
+  { .id = MCP2515_READ_RXB0SIDH, .data = MCP2515_READ_RXB0D0, .int_flag = MCP2515_CANINT_RX0IE },
+  { .id = MCP2515_READ_RXB1SIDH, .data = MCP2515_READ_RXB1D0, .int_flag = MCP2515_CANINT_RX1IE },
+};
+
+static const uint8_t s_brp_lookup[NUM_MCP2515_BITRATES] = {
+  // BRP calculated for different bitrates
+  [MCP2515_BITRATE_125KBPS] = MCP2515_CAN_BRP_125KBPS,
+  [MCP2515_BITRATE_250KBPS] = MCP2515_CAN_BRP_250KBPS,
+  [MCP2515_BITRATE_500KBPS] = MCP2515_CAN_BRP_500KBPS,
+};
+
+static Mcp2515Storage *s_mpxe_storage;
+static MxMcp2515Store s_store = MX_MCP2515_STORE__INIT;
```

Comment:
Using a static storage + store means that on MPXE there can't be more than one instance of the driver! There should definitely be one store per storage, otherwise we can't use MPXE if there's more than one MCP2515 on any board in the future.

Reply:
can we include this as a future ticket since it's nonessential to what we've got working here? also im lazy, i swear ill do it right after this merge